### PR TITLE
Make context refresh smarter

### PR DESCRIPTION
**Pull request by mvcatsifma**
It now accepts timestamp; only Resources that have been synced before
this moment are loaded from server.

### CODE REVIEW

**Comment by mvcatsifma** [comment]
@jcassee Graag een review.

File: src/resource.spec.js [hunk 1]
```diff
@@ -224,12 +224,24 @@ describe('Resource', function () {
...
+
   it('does not load a resource again if it has been synced', function () {
     resource.$syncTime = Date.now();
     resource.$load();
     expect(mockContext.httpGet).not.toHaveBeenCalled();
   });
 
+  it('does not load a resource again if it has been synced and not stale', function () {
+    resource.$syncTime = Date.now();
+    resource.$load(new Date('2016-01-01').getTime());
```
Comment (jcassee) [inline src/resource.spec.js:241]: These tests assume `Date.now()` is after 2016-01-01. Not unreasonable, but why not just use something like `new Date('2016-01-02')`?

File: src/resource.js [hunk 2]
```diff
@@ -207,11 +207,11 @@ angular.module('hypermedia')
        * @returns a promise that is resolved to the resource
        * @see Resource#$syncTime
        */
-      $load: {value: function () {
-        if (this.$syncTime) {
-          return $q.when(this);
-        } else {
+      $load: {value: function (ts) {
```
Comment (jcassee) [inline src/resource.js:212]: I think you forgot to document the new function argument. (Also in `context.js`.)

**Comment by jcassee** [comment]
Nice changeset, thanks! Just two minor things. I will look at updating the README once this PR is merged.

**Comment by jcassee** [comment]
Thanks, Thomas! I will release a new version shortly.

**Comment by jcassee** [comment]
Changes released in version 0.8.1.

**Pull request merged**